main/php_syslog.c

```diff
@@ -86,11 +86,23 @@ PHPAPI void php_syslog(int priority, const char *format, ...) /* {{{ */
 			break;
 		}
 
-		if (c != '\n')
+		/* check for NVT ASCII only unless test disabled */
+		if (((0x20 <= c) && (c <= 0x7e)))
 			smart_string_appendc(&sbuf, c);
-		else {
+		else if ((c >= 0x80) && (PG(syslog_filter) != PHP_SYSLOG_FILTER_ASCII))
+			smart_string_appendc(&sbuf, c);
+		else if (c == '\n') {
 			syslog(priority, "%.*s", (int)sbuf.len, sbuf.c);
 			smart_string_reset(&sbuf);
+		} else if ((c < 0x20) && (PG(syslog_filter) == PHP_SYSLOG_FILTER_NONE))
+			smart_string_appendc(&sbuf, c);
+		else {
+			smart_string_appendc(&sbuf, '#');
```

Comment:
Not probably a big issue but I think it might be more useful to show hex value prefixed with `\x` - seems a bit more common but I might be wrong - not sure what other syslog clients do..?

Reply:
`rsyslog` seems to use `#%03o` for formatting.  `syslog-ng` uses `\\x02x`.  Seems to go either way.

Reply:
ah cool. I see that you already converted that to hex which is great! Maybe we could use the same format as syslog-ng in that case (it's just adding extra `%` and `x` chars) so it's visible that those are the replaced characters.

Reply:
> I see that you already converted that to hex which is great! Maybe we could use the same format as syslog-ng in that case (it's just adding extra % and x chars) so it's visible that those are the replaced characters.

Sorry, I meant that `\\x%02x` is the literal printf-style format string.  So DEL would format as `\x7f` (which is also how it would appear as a literal in a C string).  What would you like to see it format as?